src/Assetic/Filter/StylusFilter.php

```diff
@@ -81,14 +92,13 @@ public function filterLoad(AssetInterface $asset)
 EOF;
 
         // parser options
-        $parserOptions = array();
         if ($dir = $asset->getSourceDirectory()) {
-            $parserOptions['paths'] = array($dir);
-            $parserOptions['filename'] = basename($asset->getSourcePath());
+            $this->parserOptions['paths'] = array($dir);
+            $this->parserOptions['filename'] = basename($asset->getSourcePath());
```

Comment:
this is wrong. Parsing a given asset must not change the state of the filter. Otherwise it will also affect subsequent assets